jcl/src/java.base/share/classes/java/lang/String.java

```diff
@@ -1794,29 +1794,33 @@ public boolean equalsIgnoreCase(String string) {
 			byte byteAtO1Last = helpers.getByteFromArrayByIndex(s1Value, s1len - 1);
 			byte byteAtO2Last = helpers.getByteFromArrayByIndex(s2Value, s1len - 1);
 
-			if (byteAtO1Last != byteAtO2Last &&
-					toUpperCase(helpers.byteToCharUnsigned(byteAtO1Last)) != toUpperCase(helpers.byteToCharUnsigned(byteAtO2Last))) {
+			if ((byteAtO1Last != byteAtO2Last)
+					&& (Character.toUpperCase(helpers.byteToCharUnsigned(byteAtO1Last)) != Character.toUpperCase(helpers.byteToCharUnsigned(byteAtO2Last)))
```

Comment:
Is this change introduced to fix an existing `equalsIgnoreCase` problem?
`private static int toUpperCase(int codePoint)` invokes `Character.toUpperCase()` internally but avoid it for case `codePoint < 128` apparently for performance reason.

Reply:
I tested locally with `toUpperCase(int codePoint)` and `Character.toUpperCase()` and the simple test case results were the same. So it seems it would be better to keep the existing `toUpperCase(int codePoint)` usage, for performance purposes.

This makes me think that we should use `Character.toLowerCase(toUpperCase(int codePoint))` in the spots where we're now using `Character.toLowerCase(Character.toUpperCase())`.

Reply:
> This makes me think that we should use Character.toLowerCase(toUpperCase(int codePoint)) in the spots where we're now using Character.toLowerCase(Character.toUpperCase()).

Yup, probably can go `toLowerCase(toUpperCase(int codePoint))`.

Reply:
In addition, it seems `String.compareToIgnoreCase()` is doing the right thing, but could you add a test for it against the `Turkish` characters reported as well?